kolibri/utils/version.py

```diff
@@ -308,37 +301,30 @@ def get_prerelease_version(version):
     if tag_describe:
 
         git_version, suffix = get_version_from_git(tag_describe)
-
-        if not git_version[:3] == version[:3]:
-            # If it's the 0th alpha, load suffix info from git changeset
-            if version[4] == 0 and version[3] == "alpha":
+        if not suffix:
+            if not git_version[:3] == version[:3]:
+                # If it's the 0th alpha, load suffix info from git changeset
+                # if version[4] == 0 and version[3] == "alpha":
                 # Throw away the description from git
                 suffix = get_git_changeset()
+
                 # Replace 'alpha' with .dev
                 return major + ".dev0" + suffix
 
-            # If the tag was not of a final version, we will fail.
-            elif not git_version[4] == "final" and git_version[:3] > version[:3]:
-                raise AssertionError(
-                    (
-                        "Version detected from git describe --tags, but it's "
-                        "inconsistent with kolibri.__version__."
-                        "__version__ is: {}, tag says: {}."
-                    ).format(str(version), git_version)
-                )
-
-        if git_version[3] == "final" and version[3] != "final":
+                # # If the tag was not of a final version, we will fail.
+        if git_version[:3] > version[:3]:
+            # If the tag was of a final version, we will use it.
             raise AssertionError(
-                "You have added a final tag without bumping kolibri.VERSION, "
-                + "OR you need to make a new alpha0 tag. Current tag: {}".format(
-                    git_version
-                )
-                + "\n\n"
-                "Often, this is because of missing tag information, try "
-                "running:\n"
-                "\n"
-                "   git fetch <upstream>"
+                (
+                    "Version detected from git describe --tags, but it's "
+                    "inconsistent with kolibri.__version__."
+                    "__version__ is: {}, tag says: {}."
+                ).format(str(version), git_version)
             )
+        if git_version[:3] == version[:3]:
+            if git_version[3] == "final":
+                return major
+        # If the tag was of a final version, we will use it.
 
         return (
```

Comment:
I think this `return` statement still need to be indented to only run `if git_version[:3] == version[:3]:`